### PR TITLE
Remove deprecated property from VMScaleSetVM properties

### PR DESCRIPTION
**Pull request by ashintre**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json [hunk 1]
```diff
@@ -7801,10 +7801,6 @@
           "type": "boolean",
           "description": "Specifies whether the latest model has been applied to the virtual machine."
         },
-        "protectFromScaleIn" : {
-          "type": "boolean",
-          "description": "Specifies whether the virtual machine instance shouldn't be considered for deletion during a scale-in operation"
-        },
```
Comment (anuchandy) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json]: Did we follow the ARM guideline for deprecating properties ?

**Review by anuchandy — COMMENTED** [review]
@ravbhatnagar PR removes a deprecated property from stable API Version. Could you review?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json (hunk 1 above)
Comment (sanjaiganesh) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json]: Yes. This must be done in new api version. Refer to 'versioning/deprecation' section under 'API contracts' of the ARM wiki. http://aka.ms/armwiki

Comment (ashintre) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json]: This should be nested inside another property. It's not usable currently as-is.

**Comment by ravbhatnagar** [comment]
@ashintre - can you please confirm service behavior if this property is used as currently modeled in swagger. If it is going to fail on service side, then I am fine approving this PR even though its a breaking change. But since it would never have worked before, we should be fine making this change. Please confirm.

**Review by anthony-c-martin — APPROVED** [review]

**Comment by anuchandy** [comment]
Response from @ashintre : "Yes, it would fail on the service side. Luckily, when I checked a week ago, our data showed no one had attempted to use it thus far"

**Review by anuchandy — APPROVED** [review]

**Pull request merged**